src/jit/gentree.h

```diff
@@ -946,7 +946,7 @@ struct GenTree
 #define GTF_IND_ARR_LEN             0x80000000 // GT_IND   -- the indirection represents an array length (of the REF
                                                //             contribution to its argument).
 #define GTF_IND_VOLATILE            0x40000000 // GT_IND   -- the load or store must use volatile sematics (this is a nop on X86)
-#define GTF_IND_NONFAULTING         0x20000000 // GT_IND   -- An indir that cannot fault.
+#define GTF_IND_NONFAULTING         0x20000000 // Operations for which OperIsIndirOrArrLength() is true  -- An indir that cannot fault.
```

Comment:
I'd recommend defining `GTF_ARRLEN_NONFAULTING` where the other `GTF_ARRLEN` flags are defined and adding a `static_assert_no_msg` that the two flags agree -- that'll make it easier not to mistakenly try to re-use that bit if somebody goes to add another flag for array length nodes in the future.  We currently do the same for `GTF_FLD_VOLATILE` and `GTF_CLS_VAR_VOLATILE` (and the static assertion is over by their use in morph that requires them to be the same).